torch/testing/_internal/test_subclasses.py

```diff
@@ -0,0 +1,53 @@
+# mypy: ignore-errors
+
+import torch
+import torch.utils._pytree as pytree
+
+
+class SubclassWithTensorFactory(torch.Tensor):
```

Comment:
this file should go in https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_subclass.py